flink-table/flink-sql-parser/src/main/java/org/apache/flink/table/calcite/RawRelDataTypeFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.calcite;
+
+import org.apache.calcite.rel.type.RelDataType;
+
+/**
+ * A factory for creating a {@link RelDataType} that describes a RAW type such as
+ * {@code RAW('org.my.Class', 'sW3Djsds...')}.
+ */
+public interface RawRelDataTypeFactory {
+
+	RelDataType createRawType(String className, String serializerString);
```

Comment:
Can we reuse the `FlinkRelDataTypeFactory`, we didn't make factory for specific rel type.

Reply:
`FlinkRelDataTypeFactory` is not accessible in `flink-sql-parser`

Reply:
Then just a singleton `RAW` type instance is enough.

Reply:
This method is also needed for `SqlRawTypeNameSpec`. How does a singleton `RAW` type instance help here?